x/bsonx/bsoncore/bsoncore_test.go

```diff
@@ -935,6 +935,12 @@ func TestNullBytes(t *testing.T) {
 			})
 		}
 	})
+	t.Run("sub document field name", func(t *testing.T) {
+		createDocFn := func() {
+			NewDocumentBuilder().StartDocument("a\x00").AppendDocument("foobar", []byte("foo")).FinishDocument()
```

Comment:
```suggestion
			NewDocumentBuilder().StartDocument("foobar").AppendDocument("a\x00", []byte("foo")).FinishDocument()
```
I _think_ this is making a document where a top-level key contains null bytes. We want to be asserting that BSON creation panics in the document builder when null bytes are in a field that is embedded.

Reply:
Ah yeah that seems right thank you